### PR TITLE
Fix residing agent memory

### PR DESCRIPTION
**Pull request by BjoernLudwigPTB**
We spotted a left over from the times of `self.memory` in _agents.py_ and went to correct that. Then we stumbled across some minor issues in old docstrings and a loop over a dictionary via the key, where we could simply loop over keys and values. Hope this breaks nothing.

### CODE REVIEW

**Comment by BjoernLudwigPTB** [comment]
@bangxiangyong This is ready to be review by you, as soon as you find the time, ideally before friday.

File: agentMET4FOF/agents.py [hunk 1]
```diff
@@ -1654,22 +1693,23 @@ def on_received_message(self, message):
         """
         Handles incoming data from 'default' and 'plot' channels.
 
-        Stores 'default' data into `self.memory` and 'plot' data into `self.plots`
+        Stores 'default' data into :attr:``self.buffer`` and 'plot' data into
```
Comment (BjoernLudwigPTB) [inline agentMET4FOF/agents.py:1696]:
```suggestion
        Stores 'default' data into :attr:`buffer` and 'plot' data into
```

File: agentMET4FOF/agents.py [hunk 2]
```diff
@@ -1654,22 +1693,23 @@ def on_received_message(self, message):
         """
         Handles incoming data from 'default' and 'plot' channels.
 
-        Stores 'default' data into `self.memory` and 'plot' data into `self.plots`
+        Stores 'default' data into :attr:`buffer` and 'plot' data into
+        :attr:``self.plots``
```
Comment (BjoernLudwigPTB) [inline agentMET4FOF/agents.py:1697]:
```suggestion
        :attr:`plots`
```

**Comment by bangxiangyong** [comment]
Great, i like this improvement in the code !

**Pull request merged**